### PR TITLE
refactor(exporter-file) Change load configuration

### PR DESCRIPTION
**Pull request by wfajczyk**
# Description

Refactor Export, add taging to configuration

Issue #1542 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [ ] I have read the contribution requirements and fulfill them.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: UPGRADE-1.2.md [hunk 1]
```diff
@@ -5,3 +5,11 @@ Core\Test:
     * For the context to work it does require BehatchBridgeExtension enabled in Behat configuration
 * The signature of constructor `ApiAuthTokenContext` has changed
     * For the context to work it does require BehatchBridgeExtension enabled in Behat configuration
+
+ExporterFile:
+* The signature of constructor `StartProcessCommandHandler::__construct(ExportRepositoryInterface $repository, TempFileStorage $storage, ExportProductBuilder $productBuilder, ExportTemplateElementBuilder $templateElementBuilder, ExportAttributeBuilder $attributeBuilder, ExportOptionBuilder $optionBuilder, ExportCategoryBuilder $categoryBuilder, ExportTemplateBuilder $templateBuilder)`
```
Comment (piotrkreft) [inline UPGRADE-1.2.md:10]: Just stick with the information that the constructor has changed(as in UPGRADE 1.1.md)

File: UPGRADE-1.2.md [hunk 2]
```diff
@@ -5,3 +5,11 @@ Core\Test:
...
+  has been changed to `StartProcessCommandHandler::__construct(ExportRepositoryInterface $repository, TempFileStorage $storage, iterable $builders)` 
+* The signature of constructor `ProcessExportCommandHandler::__construct(ChannelRepositoryInterface $channelRepository, ExportRepositoryInterface $exportRepository, CommandBusInterface $commandBus, array $steps)`
+  has been changed to `ProcessExportCommandHandler::__construct(ChannelRepositoryInterface $channelRepository, ExportRepositoryInterface $exportRepository, CommandBusInterface $commandBus, iterable $steps)` 
+* Interface `ExportHeaderBuilderInterface` has been added
```
Comment (piotrkreft) [inline UPGRADE-1.2.md:14]: Not a BC Break

File: UPGRADE-1.2.md [hunk 3]
```diff
@@ -5,3 +5,11 @@ Core\Test:
...
+* Classes `ExportProductBuilder`, `ExportTemplateElementBuilder`, `ExportAttributeBuilder`, `ExportOptionBuilder`, `ExportCategoryBuilder`, `ExportTemplateBuilder` implements `ExportHeaderBuilderInterface`
```
Comment (piotrkreft) [inline UPGRADE-1.2.md:15]: Not a BC Break

File: module/exporter-file/tests/Infrastructure/Handler/Export/StartProcessCommandHandlerTest.php [hunk 4]
```diff
@@ -43,12 +43,14 @@ public function testHandling(): void
...
             $exportRepository,
             $storage,
-            $productBuilder,
-            $elementBuilder,
-            $attributeBuilder,
-            $optionBuilder,
-            $categoryBuilder,
-            $templateBuilder
+            [
+                $productBuilder,
```
Comment (piotrkreft) [inline module/exporter-file/tests/Infrastructure/Handler/Export/StartProcessCommandHandlerTest.php:47]: I guess test could be decoupled from implementations now as well

File: module/exporter-file/src/Resources/config/services.yml [hunk 5]
```diff
@@ -34,36 +34,46 @@ services:
...
-            $steps:
-                - '@Ergonode\ExporterFile\Infrastructure\Processor\Step\AttributeExportProcessorStep'
-                - '@Ergonode\ExporterFile\Infrastructure\Processor\Step\OptionExportProcessorStep'
-                - '@Ergonode\ExporterFile\Infrastructure\Processor\Step\CategoryExportProcessorStep'
-                - '@Ergonode\ExporterFile\Infrastructure\Processor\Step\ProductExportProcessStep'
-                - '@Ergonode\ExporterFile\Infrastructure\Processor\Step\TemplateExportProcessorStep'
+            $steps: !tagged_iterator 'export-file.steps_process'
 
     Ergonode\ExporterFile\Infrastructure\Builder\ExportTemplateBuilder:
+        tags: ['export-file.header_builder']
```
Comment (piotrkreft) [inline module/exporter-file/src/Resources/config/services.yml:52]: Global tagging added in extension. Should be redundant

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by rprzedzik — APPROVED** [review]

**Pull request merged**